repose-aggregator/services/service-client/impl/src/main/java/org/openrepose/core/services/serviceclient/akka/impl/AkkaServiceClientImpl.java

```diff
@@ -72,27 +74,42 @@ public AkkaServiceClientImpl(HttpClientService httpClientService) {
                 .expireAfterWrite(FUTURE_CACHE_TTL, FUTURE_CACHE_UNIT)
                 .build();
 
-
-        tokenActorRef = actorSystem.actorOf(new Props(new UntypedActorFactory() {
-            public UntypedActor create() {
-
-                return new AuthTokenFutureActor(serviceClient);
-            }
-        }).withRouter(new RoundRobinRouter(numberOfActors)), "authRequestRouter");
+        createTokenActor();
     }
 
-
     @PreDestroy
     public void destroy() {
-        //Tie this actor system to our bean lifecycle
+        // tie this actor system to our bean lifecycle
         actorSystem.shutdown();
     }
 
     @Override
+    public void setConnectionPoolId(String connectionPoolId) {
+        if (StringUtils.equals(this.connectionPoolId, connectionPoolId)) {
+            return;
+        }
+
+        ActorRef oldTokenActorRef = tokenActorRef;
+        this.connectionPoolId = connectionPoolId;
+        createTokenActor();
+
+        if (oldTokenActorRef != null) {
+            actorSystem.stop(oldTokenActorRef);
+        }
+    }
+
+    private void createTokenActor() {
+        ServiceClient serviceClient = new ServiceClient(connectionPoolId, httpClientService);
```

Comment:
The actor can be created giving a reference to the HttpClient Service, and a default ConnectionPoolId.

Then you can just issue it messages and have it create it's own service client depending on whatever connection ID it needs.